### ISSUE TITLE
onMapReadyCallback never called

### ISSUE THREAD

**Issue by vicobz** [mapbox/mapbox-gl-native#11094]
Hello,

After few months of dev using this amazing Mapbox SDK for Android, I've been facing a bug since 2 days and I really don't know how to handle it. 

The onMapReadyCallback gets never called on my OnePlus 5, while there has been no problem on this feature during the last months. As this is the enter point to get a MapBoxMap object, I'm kind of stuck here.

**Platform:** OnePlus 5 - Android 8.0.0 Oreo
**Mapbox SDK version:** 5.3.2

### Steps to trigger behavior

 Call getMapAsync(this) in activity's onCreate: 

```java
 protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_map);
        ButterKnife.bind(this);
        mapView.onCreate(savedInstanceState);
        mapView.getMapAsync(this);
        mapView.addOnMapChangedListener(change -> Log.i(TAG, "on map changed: " + change));
    } 
```

### Expected behavior
The onMapReady callback gets called and so I can hook my MapBoxMap object.

### Actual behavior
onMapReady is never called, although 

> DID_FINISH_RENDERING_MAP_FULLY_RENDERED

event seems to be called as its value (13) is printed from the onMapChanged listener above.

If anyone has a clue on this, I would be really thankful. This is quite urgent too :)

**Comment by tobrun** [comment]
@vicobz The signal for OnMapReady to be invoked is actually not `DID_FINISH_RENDERING_MAP_FULLY_RENDERED` but `DID_FINISH_LOADING_STYLE` with value 14. 

Could you add a little bit more information on how you are adding your style? Are you hooking into all the required lifecycle methods? eg. onStart/onResume?

**Comment by vicobz** [comment]
@tobrun Thanks for your reply, didn't see the notification yet.

Yes I did hook all the required lifecycle methods. Finally I think this issue was due to an initialization trouble occurring while instantiating a CustomNavigationNotification object for MapboxNavigation initialization. 

Some part of the code was blocking while trying to set notification channel configuration depending on OS version, aiming to avoid verbose Android SDK Logs about unset channel configuration on Android O. I didn't have time to investigate on this issue, will try to add more info asap if needed ! :)
